src/test/shared/defaultAwsContext.test.ts

```diff
@@ -25,10 +29,115 @@ describe('DefaultAwsContext', () => {
         public async writeSetting<T>(
             settingKey: string, value: T | undefined,
             target: ConfigurationTarget
-        ): Promise<void> {
-        }
+        ): Promise<void> { }
     }
 
+    it('gets credentials if a profile exists with credentials', async () => {
+
+        class TestConfiguration extends ContextTestsSettingsConfigurationBase {
+            public readSetting<T>(settingKey: string, defaultValue?: T): T | undefined {
+                if (settingKey === profileSettingKey) {
+                    return testProfileValue as any as T
+                }
+
+                return super.readSetting(settingKey, defaultValue)
+            }
+        }
+
+        const reportedCredentials = new Credentials(testAccessKey, testSecretKey)
+
+        class TestCredentialsManager extends CredentialsManager {
+            public async getCredentials(profileName: string): Promise<Credentials> {
+                return reportedCredentials
+            }
+        }
+
+        const testContext = new DefaultAwsContext(
+            new TestConfiguration(),
+            new FakeExtensionContext(),
+            new TestCredentialsManager()
+        )
+        const creds = await testContext.getCredentials()
+        assert.strictEqual(creds, reportedCredentials)
+    })
+
+    it('gets credentials if a profile exists with credentials', async () => {
+        const overrideProfile = 'asdf'
+
+        class TestConfiguration extends ContextTestsSettingsConfigurationBase {
```

Comment:
You can remove duplication of code by defining `TestConfiguration` once and taking in the value you wish to return in the constructor. This will reduce LOC in each test and help improve understanding what each test is trying to accomplish.